lib/driver.js

```diff
@@ -321,15 +322,17 @@ class YouiEngineDriver extends BaseDriver {
   }
 
   startYIRokuSession (caps) {
-    logger.info("Launching app");
+    logger.info("Launching Roku app");
+    this.validateAppLocation(caps.app);
     let shell = require('shelljs');
     let roku_install_script = "curl -v -# -f -i --user '" + caps.username + ":" + caps.password + "' --digest --progress-bar -F 'mysubmit=Install' -F 'archive=@" + caps.app + "' -F 'passwd=' http://" + caps.youiEngineAppAddress + "/plugin_install | grep '<font color' | sed 's/<font color=\'red\'>//' "; // eslint-disable-line no-useless-escape
     shell.exec(roku_install_script);
     //"curl -v -# -f -i --user 'rokudev:youi' --digest --progress-bar -F 'mysubmit=Install' -F 'archive=@/Users/simongranger/Downloads/VideoPlayer.bin' -F 'passwd=' http://10.100.89.245/plugin_install | grep '<font color' | sed 's/<font color=\'red\'>//' "
   }
 
   startYITVOSSession (caps) {
-    logger.info("Launching app");
+    logger.info("Launching tvOS app");
```

Comment:
Should you not modify all of the platform-specific sessions in a similar manner? And not just osx/tvos/roku?

Reply:
These are the only ones we launch from within our driver. iOS and Android apps are launched via the proxied drivers.

Reply:
I see Paige's point. I'm not sure you shouldn't call this once to validate the app location just once. If the app isn't found at the location, then it's going to fail whether we deploy it or not no?